File: crop_logos.js
import { Jimp } from 'jimp';
import path from 'path';
import fs from 'fs';

async function run() {
  const imgPath = "C:/Users/ASUS/.gemini/antigravity-ide/brain/tempmediaStorage/media__1780910144951.png";
  console.log("Loading image buffer...");
  const buffer = fs.readFileSync(imgPath);
  const image = await Jimp.read(buffer);
  const width = image.bitmap.width;
  const height = image.bitmap.height;
  console.log(`Image dimensions: ${width}x${height}`);

  // We want to find a row index that separates the two logos.
  // The first logo is "Kemenkes RS Nusantara".
  // The second logo is "Bali International Hospital".
  // Let's find a row in the middle 30% to 70% height range that is empty/white.
  let splitY = Math.floor(height / 2);
  let bestRow = splitY;
  let minDiff = Infinity;

  // Let's find a row with minimum variance or closest to pure white / transparent
  for (let y = Math.floor(height * 0.35); y < Math.floor(height * 0.65); y++) {
    let nonWhiteCount = 0;
    for (let x = 0; x < width; x++) {
      const pixelColor = image.getPixelColor(x, y); // RGBA hex number
      const r = (pixelColor >> 24) & 0xff;
      const g = (pixelColor >> 16) & 0xff;
      const b = (pixelColor >> 8) & 0xff;
      const a = pixelColor & 0xff;
      
      // If it is not white (R>245, G>245, B>245) and not transparent (A<10)
      if (!(a < 10 || (r > 245 && g > 245 && b > 245))) {
        nonWhiteCount++;
      }
    }
    if (nonWhiteCount < minDiff) {
      minDiff = nonWhiteCount;
      bestRow = y;
    }
  }

  console.log(`Determined split row at Y = ${bestRow} with non-white pixel count = ${minDiff}`);

  // Crop top logo
  const topImg = image.clone().crop({ x: 0, y: 0, w: width, h: bestRow });
  // Crop bottom logo
  const bottomImg = image.clone().crop({ x: 0, y: bestRow, w: width, h: height - bestRow });

  // Let's trim function (simple bounding box search)
  function getBBox(img) {
    const w = img.bitmap.width;
    const h = img.bitmap.height;
    let minX = w, maxX = 0, minY = h, maxY = 0;
    let found = false;

    for (let y = 0; y < h; y++) {
      for (let x = 0; x < w; x++) {
        const color = img.getPixelColor(x, y);
        const r = (color >> 24) & 0xff;
        const g = (color >> 16) & 0xff;
        const b = (color >> 8) & 0xff;
        const a = color & 0xff;

        // If not white/transparent, it's content
        if (!(a < 10 || (r > 250 && g > 250 && b > 250))) {
          if (x < minX) minX = x;
          if (x > maxX) maxX = x;
          if (y < minY) minY = y;
          if (y > maxY) maxY = y;
          found = true;
        }
      }
    }

    if (!found) return null;
    // Add 5px padding around bounding box if possible
    minX = Math.max(0, minX - 5);
    minY = Math.max(0, minY - 5);
    maxX = Math.min(w - 1, maxX + 5);
    maxY = Math.min(h - 1, maxY + 5);
    return { x: minX, y: minY, w: maxX - minX + 1, h: maxY - minY + 1 };
  }

  const topBBox = getBBox(topImg);
  const finalTop = topBBox ? topImg.clone().crop(topBBox) : topImg;

  const bottomBBox = getBBox(bottomImg);
  const finalBottom = bottomBBox ? bottomImg.clone().crop(bottomBBox) : bottomImg;

  // Save them
  await finalTop.write("e:/Fani/Semester Kerja/DPSA/webDPSA/public/images/client/RS Nusantara.png");
  await finalBottom.write("e:/Fani/Semester Kerja/DPSA/webDPSA/public/images/client/Bali International Hospital.png");
  console.log("Logos saved successfully!");
}

run().catch(err => {
  console.error(err);
});

File: crop_logos.py
from PIL import Image

# Open the uploaded image
img_path = r"C:/Users/ASUS/.gemini/antigravity-ide/brain/tempmediaStorage/media__1780910144951.png"
img = Image.open(img_path)
width, height = img.size
print(f"Image dimensions: {width}x{height}")

# Let's crop the image. It contains two logos stacked vertically.
# We will inspect where to crop. Let's crop top half and bottom half first.
# Or we can scan for horizontal lines of fully transparent/white pixels to find the gap.

# Convert to RGBA
img_rgba = img.convert("RGBA")

# Let's find rows with very little color or just white/transparent to locate the separation point.
# The background looks white.
import numpy as np
data = np.array(img_rgba)

# We want to find a row index that is mostly white/transparent.
# Let's check each row to see if it's blank (either transparent alpha=0 or white R=255, G=255, B=255)
is_row_blank = []
for y in range(height):
    row = data[y]
    # Check if all pixels in the row are close to white (e.g. R>250, G>250, B>250) or alpha close to 0
    blank_pixels = 0
    for x in range(width):
        r, g, b, a = row[x]
        if a < 10 or (r > 245 and g > 245 and b > 245):
            blank_pixels += 1
    is_row_blank.append(blank_pixels == width)

# Let's find the longest run of blank rows in the middle section (e.g., between 35% and 65% of height)
mid_start = int(height * 0.35)
mid_end = int(height * 0.65)
blank_in_middle = [y for y in range(mid_start, mid_end) if is_row_blank[y]]

if blank_in_middle:
    split_y = blank_in_middle[len(blank_in_middle) // 2]
    print(f"Automatically detected split at y={split_y}")
else:
    split_y = height // 2
    print(f"Could not automatically detect split, using middle: y={split_y}")

# Crop RS Nusantara (Top)
rs_nusantara = img.crop((0, 0, width, split_y))
# Crop Bali International Hospital (Bottom)
bali_hospital = img.crop((0, split_y, width, height))

# Let's trim whitespace for both cropped images
def trim(im):
    # Trim white/transparent borders
    bg = Image.new(im.mode, im.size, (255, 255, 255, 255) if im.mode == "RGBA" else im.getpixel((0,0)))
    diff = ImageChops.difference(im, bg)
    diff = ImageChops.add(diff, diff, 2.0, -100)
    bbox = diff.getbbox()
    if bbox:
        return im.crop(bbox)
    return im

from PIL import ImageChops
try:
    rs_nusantara_trimmed = trim(rs_nusantara)
    bali_hospital_trimmed = trim(bali_hospital)
except Exception as e:
    print(f"Error during trim: {e}")
    rs_nusantara_trimmed = rs_nusantara
    bali_hospital_trimmed = bali_hospital

# Save the images
rs_nusantara_trimmed.save(r"e:/Fani/Semester Kerja/DPSA/webDPSA/public/images/client/RS Nusantara.png")
bali_hospital_trimmed.save(r"e:/Fani/Semester Kerja/DPSA/webDPSA/public/images/client/Bali International Hospital.png")
print("Saved successfully!")
